chain/network/src/peer_manager/peer_store.rs

```diff
@@ -56,56 +56,82 @@ impl PeerStore {
+                    match addr_2_peer.entry(peer_addr) {
                         Entry::Occupied(entry) => {
                             // There is already a different peer_id with this address.
                             error!(target: "network", "Two boot nodes have the same address {:?}", entry.key());
+                            // TODO: add panic!, `boot_nodes` list is wrong
                         }
                         Entry::Vacant(entry) => {
                             entry.insert(VerifiedPeer::signed(peer_info.id.clone()));
-                            peer_states.insert(
+                            peerid_2_state.insert(
                                 peer_info.id.clone(),
-                                KnownPeerState::new(peer_info.clone()),
+                                KnownPeerState::new(peer_info.clone(), now),
                             );
                         }
                     }
                 }
             }
-        }
+            else {
+                // TODO: add panic!, `boot_nodes` list is wrolng
+
+            }
+        });
 
         let now = to_timestamp(Utc::now());
         for (key, value) in store.iter(ColPeers) {
             let peer_id: PeerId = PeerId::try_from_slice(key.as_ref())?;
-            let mut peer_state: KnownPeerState = KnownPeerState::try_from_slice(value.as_ref())?;
+            let peer_state: KnownPeerState = KnownPeerState::try_from_slice(value.as_ref())?;
             // Mark loaded node last seen to now, to avoid deleting them as soon as they are loaded.
-            peer_state.last_seen = now;
-            match peer_state.status {
-                KnownPeerStatus::Banned(_, _) => {}
-                _ => peer_state.status = KnownPeerStatus::NotConnected,
+
+            let peer_state = KnownPeerState {
+                peer_info: peer_state.peer_info,
+                first_seen: peer_state.first_seen,
+                last_seen: now,
+                status: match peer_state.status {
+                    banned_status @ KnownPeerStatus::Banned(_, _) => banned_status,
+                    _ => KnownPeerStatus::NotConnected,
+                },
             };
 
-            if let Some(current_peer_state) = peer_states.get_mut(&peer_id) {
-                // This peer is a boot node and was already added so skip.
-                if peer_state.status.is_banned() {
-                    current_peer_state.status = peer_state.status;
+            match peerid_2_state.entry(peer_id) {
+                // Peer is a boot node
+                Entry::Occupied(mut current_peer_state) => {
+                    if peer_state.status.is_banned() {
+                        // If it says in database, that peer should be banned, ban the peer.
+                        current_peer_state.get_mut().status = peer_state.status;
+                    }
+                    // Maybe a bug:
+                    // For boot node, it's first_seen = now
```

Comment:
what is first_seen used for? (from what I've seen in the code, seems that we're never reading this value)